mne/morph.py

```diff
@@ -0,0 +1,953 @@
+            self.vol_info['mri_to']['zooms'] = mri_to.header.get_zooms()
+            self.vol_info['mri_to']['affine'] = mri_to.affine
+
+            # get source space meta data
+            shape = src[0]['shape']
+            self.src_info['shape'] = (shape[2], shape[1], shape[0])
+            self.src_info['zooms'] = tuple(np.asanyarray(
+                self.vol_info['mri_from']['shape'][:3]).astype(
+                'float') / np.asanyarray(self.src_info['shape'][:3]))
+            self.src_info['zooms_mri'] = self.vol_info['mri_from'][
+                'zooms']
+            self.src_info['affine'] = src[0]['vox_mri_t']['trans']
+            self.src_info['mri_ras_t'] = src[0]['mri_ras_t']['trans']
+            self.src_info['interpolator'] = src[0]['interpolator']
+            self.src_info['inuse'] = src[0]['inuse']
+
+            # pre-compute morph
+            self.morph_data = _compute_morph_sdr(mri_from, mri_to,
+                                                 **self.options)
+        # SourceEstimate | MixedSourceEstimate
+        elif self.kind == 'surface':
+            logger.info('surface source space inferred...')
+
+            # get surface data
+            if data_from is None:
+                data_from = []
+                for s in src:
+                    data_from.append(s['vertno'])
+
+            if data_to is None and subject_to == 'fsaverage':
+                data_to = [np.arange(10242)] * len(data_from)
+
+            # pre-compute morph
+            self.morph_data = _compute_morph_matrix(
+                subject_from=self.subject_from,
+                subject_to=self.subject_to,
+                vertices_from=data_from,
+                vertices_to=data_to,
+                subjects_dir=self.subjects_dir,
+                **self.options)
+            self.src_info['vertno'] = data_to
+        else:
+            return
+
+    def __call__(self, stc_from):
+        """Morph data.
+
+        Parameters
+        ----------
+        stc_from : VolSourceEstimate | SourceEstimate | VectorSourceEstimate
+            The source estimate to morph.
+
+        Returns
+        -------
+        stc_to : VolSourceEstimate | SourceEstimate | VectorSourceEstimate
+            The morphed source estimate.
+        """
+        if stc_from.subject is not None \
+                and stc_from.subject != self.subject_from:
+            raise ValueError('stc_from.subject and '
+                             'morph.subject_from must match')
+
+        stc_to = _apply_morph(stc_from, self)
+
+        if self.kind == 'volume':
+            _update_src_vol_info(self, self.morph_data)
+
+        stc_to.subject = self.subject_to
+        return stc_to
+
+    @verbose
+    def save(self, fname, verbose=None):
+        """Save the morph for source estimates to a file.
+
+        Parameters
+        ----------
+        fname : str
+            The stem of the file name. '-morph.h5' will be added
```

Comment:
The docstring does not explain what is actually done. `'-morph.h5` is not always added